test/test_nn.py

```diff
@@ -3529,6 +3529,27 @@ def test_adaptive_pooling_avg_nhwc(self):
         self.assertEqual(out, ref_out)
         self.assertEqual(input.grad, ref_input.grad)
 
+    @unittest.skipIf(not TEST_CUDA, "CUDA unavailable")
```

Comment:
why are you testing only cuda here? There's no implementation on cpu?
Can you please move your tests to the generic test case (class TestNNDeviceType, on line 8575 approx) and use generic framework for them https://github.com/pytorch/pytorch/wiki/Writing-tests-that-run-on-all-available-device-types

Reply:
No there's no CPU NHWC support yet.
Will move and update decorator as instructed.